.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -2467,72 +2467,87 @@ stages:
-        endpoint="/test/session/start?test_session_token=$token"
-        echo "Starting snapshot session with $endpoint"
-        docker-compose -p ddtrace_$(Build.BuildNumber) exec -T test-agent /usr/bin/curl --fail "http://localhost:8126$endpoint"
-      displayName: start snapshot session
-      
-    - script: |
-        docker-compose -p ddtrace_$(Build.BuildNumber) run -e dockerTag=$(dockerTag) smoke-tests
-      env:
-        dockerTag: $(dockerTag)
-      displayName: docker-compose run smoke-tests
-
-    - script: |
-        token=$(token)
-        endpoint="/test/session/traces?test_session_token=$token"
-        echo "Dumping traces with $endpoint"
-        docker-compose -p ddtrace_$(Build.BuildNumber) exec -T test-agent /usr/bin/curl -o /debug_snapshots/smoke_test_traces.json "http://localhost:8126$endpoint"
-
-        endpoint="/test/session/stats?test_session_token=$token"
-        echo "Dumping stats with $endpoint"
-        docker-compose -p ddtrace_$(Build.BuildNumber) exec -T test-agent /usr/bin/curl -o /debug_snapshots/smoke_test_stats.json "http://localhost:8126$endpoint"
-
-        endpoint="/test/session/requests?test_session_token=$token"
-        echo "Dumping all requests with $endpoint"
-        docker-compose -p ddtrace_$(Build.BuildNumber) exec -T test-agent /usr/bin/curl -o /debug_snapshots/smoke_test_requests.json "http://localhost:8126$endpoint"
-
-        if [ "$(publishFramework)" = "netcoreapp2.1" ]; then
-          snapshotfile="smoke_test_snapshots_2_1"
-        else
-          snapshotfile="smoke_test_snapshots"
-        fi
-
-        # Fail if the snapshots don't match
-        endpoint="/test/session/snapshot?test_session_token=$token&file=/snapshots/$snapshotfile"
-        echo "Verifying snapshot session with $endpoint"
-        docker-compose -p ddtrace_$(Build.BuildNumber) exec -T test-agent /usr/bin/curl --fail "http://localhost:8126$endpoint"
-      displayName: check snapshots
+    - template: steps/run-snapshot-test.yml
+      parameters:
+        target: 'smoke-tests'
+        snapshotPrefix: "smoke_test"
 
-    - task: DockerCompose@0
-      displayName: dump docker-compose logs for test-agent
-      inputs:
-        containerregistrytype: Container Registry
-        dockerComposeCommand: logs test-agent
-        projectName: ddtrace_$(Build.BuildNumber)
+    - publish: tracer/build_data
+      artifact: installer-smoke-test-logs_$(dockerTag)_$(System.JobAttempt)
       condition: succeededOrFailed()
       continueOnError: true
 
-    - task: DockerCompose@0
-      displayName: docker-compose stop services
+    - template: steps/run-in-docker.yml
+      parameters:
+        build: true
+        baseImage: alpine
+        command: "CheckBuildLogsForErrors"
+
+- stage: nuget_installer_smoke_tests
+  condition: and(succeeded(), eq(variables['isScheduledBuild'], 'False'))
+  dependsOn: [dotnet_tool, package_windows, generate_variables, master_commit_id]
```

Comment:
Why do we have a dependency on dotnet_tool?

Reply:
The dotnet_tool is where we build the distribution NuGet (as it depends on building _everything_, for all platforms)